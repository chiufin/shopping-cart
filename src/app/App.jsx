import React, { Component } from 'react'
import { Route, Switch } from 'react-router'
import { withRouter } from 'react-router-dom'
import PropTypes from 'prop-types'
import { connect } from 'react-redux'

import { loadApp } from '../actions/appActions'
import MainLayout from '../layouts/MainLayout'
import { HomeContainer } from '../containers'
import { homeRoute } from '../routes'

import './App.scss'

export class App extends Component {
  componentDidMount() {
    this.props.loadApp()
  }

  render() {
    return (
      <MainLayout>
        <Switch>
          <Route exact path={homeRoute.path} component={HomeContainer} />
        </Switch>
      </MainLayout>
    )
  }
}

App.propTypes = {
  loadApp: PropTypes.func,
}

export default withRouter(connect(null, {
  loadApp,
})(App))
