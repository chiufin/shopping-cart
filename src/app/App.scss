@import "./reset";
@import "../variables.scss";

body {
  position: relative;
  color: $color-black;
  background-color: #fff;
  font-family: "ArgosBook", Arial, sans-serif;
}

a {
  text-decoration: none;
  color: $color-black;

  &:hover {
    text-decoration: underline;
  }
}

img {
  max-width: 100%;
  height: auto;
  display: block;
  margin: 0 auto;
}

h1, :global(.h1) {
  font-size: 30px;
  margin: 30px 0;
  text-align: center;
}

h2, :global(.h2){
  font-size: 28px;
  margin-bottom: 30px;
  
  &:global(.header) {
    font-size: 22px;
    margin: 0;
  }
}

p {
  margin-bottom: 20px;
}

strong {
  font-weight: bold;
}

