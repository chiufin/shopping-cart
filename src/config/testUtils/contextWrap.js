import { BrowserRouter } from 'react-router-dom'
import { shallow, mount } from 'enzyme'
import { shape } from 'prop-types'

const router = {
  history: new BrowserRouter().history,
  route: {
    location: {},
    match: {},
  },
}

const createContext = () => ({
  context: { router },
  childContextTypes: {
    router: shape({}),
  },
})

// used for testing router wrapped components when content is need to Mount render
export function mountWrap(node) {
  return mount(node, createContext())
}

// used for testing router wrapped components when content is need to Shallow render
export function shallowWrap(node) {
  return shallow(node, createContext())
}
