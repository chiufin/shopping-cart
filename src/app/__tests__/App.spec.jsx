import React from 'react'
import { shallowWrap } from '../../config/testUtils/contextWrap'
import { App } from '../App'

const defaultProps = {
  loadApp: jest.fn(),
}

describe('App.jsx Mocking', () => {
  it('App Mocks correctly', () => {
    const mockApp = shallowWrap(<App {...defaultProps} />)
    expect(mockApp).toHaveLength(1)
    expect(mockApp).toMatchSnapshot()
  })
})
