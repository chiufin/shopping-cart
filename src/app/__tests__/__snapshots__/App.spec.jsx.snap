// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App.jsx Mocking App Mocks correctly 1`] = `
<MainLayout>
  <Switch>
    <Route
      component={[Function]}
      exact={true}
      path="/"
    />
  </Switch>
</MainLayout>
`;
